examples/crystal_volume/crystal_state.py

```diff
@@ -0,0 +1,90 @@
+from typing import Tuple, Optional, Sequence
+
+from rlmolecule.sql import hash_to_integer
+from rlmolecule.tree_search.graph_search_state import GraphSearchState
+from rlmolecule.tree_search.metrics import collect_metrics
+
+
+class CrystalState(GraphSearchState):
+    """
+    A State implementation which uses simple transformations (such as adding a bond) to define a
+    graph of molecules that can be navigated.
+    
+    Crystals are generated only at the final state
+    """
+    def __init__(
+        self,
+        elements: Tuple,
+        action_node: any,
+        builder: any,
+        composition: Optional[str] = None,
+        comp_type: Optional[str] = None,
+        terminal: bool = False,
+        #structure: any = None,
+    ) -> None:
+        """
+        :param builder: A CrystalBuilder class
+        :param action_node: A representation of the current state in one of the action graphs
+        :param terminal: Whether this state is a decoration of a specific structure (i.e., final state)
+        """
+        self._builder: any = builder
+        self._elements: str = elements
+        self._composition: str = composition
+        self._comp_type: str = comp_type
```

Comment:
If we had the composition we should be able to fill these other two out, right?

Reply:
Yes. At the beginning there are element combinations that are not yet compositions, but I can handle those cases easily enough